File: Solutions/F_Digit_Frequency.c
#include <stdio.h>
#include <string.h>

int main() {

    char s[1001];  

    int freq[10] = {0}; 

    scanf("%s", s);

    for (int i = 0; i < strlen(s); i++) {

        char ch = s[i];

        if (ch >= '0' && ch <= '9') {  
            int digit = ch - '0';      
            freq[digit]++;   
          
        }
    }

    for (int i = 0; i < 10; i++) {
        printf("%d ", freq[i]);

    }

    return 0;
}

File: Solutions/F_Digit_Frequency.java
import java.util.*;

public class F_Digit_Frequency {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        String s = sc.nextLine();
        int[] freq = new int[10];

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);

            if (Character.isDigit(ch)) {
                int digit = ch - '0';  
                freq[digit]++;  
            }
        }

        for (int i = 0; i < 10; i++) {
            System.out.print(freq[i] + " ");
        }

        sc.close();
    }
}
